### PR TITLE
Add private data request support and fix all cpplint errors and static analysis errors

### PR DESCRIPTION
**Pull request by michaellawes**

### CODE REVIEW

**Comment by fhanau** [comment]
Thanks for your pull request, looks like a lot of progress.
A few notes:
- This is a lot of changes, it might be better to split it up into several pull requests for easier reviewing in the future. For now, looking at it one commit at a time works.
- Linting and static analysis do not need to be fully fixed, especially at this stage. When running on mac, I still see many static analyzer issues (i.e. with `code-analyze`). We should also re-enable the runtime/references linter warning, it will warn when non-const references are being used as pass-by-reference behaves differently then pass-by-value when the variable is being modified which can lead to subtle bugs. This might result in some duplicated work, but this change should still  cut down the number of linting issues.

**Comment by fhanau** [comment]
In terms of static analysis, I'll work on fixing some of the issues after this branch has been reviewed and merged

**Comment by fhanau** [comment]
Michael, the pull request is pretty big as is, perhaps you can make future changes on a different branch?

**Comment by michaellawes** [comment]
Yeah is it alright if I merge or should i just make a new branch now?On Oct 29, 2022, at 10:53 AM, Felix Hanau ***@***.***> wrote:﻿
Michael, the pull request is pretty big as is, perhaps you can make future changes on a different branch?

**Comment by fhanau** [comment]
Ideally someone should review it beforehand, maybe Songheng or Alex can, otherwise I could take a look in the afternoon. For now just use another branch.

File: src/client/requester.cpp
Comment (fhanau) [inline src/client/requester.cpp:246]: Some code clones here, we should refactor this section later on, I'll make an issue.

**Review by fhanau — APPROVED** [review]

**Pull request merged**
